src/main/java/address/sync/cloud/CloudRateLimitStatus.java

```diff
@@ -97,7 +98,16 @@ void setQuotaResetTime(long quotaResetTime) {
         this.quotaReset = quotaResetTime;
     }
 
-    public void useQuota(int amount) {
+    public void useQuota(int amount) throws RejectedExecutionException {
+        if (quotaRemaining < amount) throw new RejectedExecutionException("No more API quota");
         quotaRemaining -= amount;
     }
+
+    public void useQuota() throws RejectedExecutionException {
+        useQuota(1);
+    }
+
+    public void replenishQuota() {
```

Comment:
Shouldnt need this. Old mechanism to deduct quota should work, but this object needs to be volatile.